tests/test_subtractTask.py

```diff
@@ -67,6 +68,41 @@ def test_mismatched_template(self):
         with self.assertRaises(AssertionError):
             task.run(template, science, sources)
 
+    def test_incomplete_template_coverage(self):
+        noiseLevel = 1.
+        border = 20
+        xSize = 400
+        ySize = 400
+        science, sources = makeTestImage(psfSize=3.0, noiseLevel=noiseLevel, noiseSeed=6, nSrc=50,
+                                         xSize=xSize, ySize=ySize)
+        template, _ = makeTestImage(psfSize=2.0, noiseLevel=noiseLevel, noiseSeed=7, nSrc=50,
+                                    templateBorderSize=border, doApplyCalibration=True,
+                                    xSize=xSize, ySize=ySize)
+
+        science_height = science.getBBox().getDimensions().getY()
+
+        def _run_and_check_coverage(template_coverage):
+            template_cut = template.clone()
+            template_height = int(science_height*template_coverage + border)
+            template_cut.image.array[:, template_height:] = 0
+            template_cut.mask.array[:, template_height:] = template_cut.mask.getPlaneBitMask('NO_DATA')
+            config = subtractImages.AlardLuptonSubtractTask.ConfigClass()
+            if template_coverage < config.requiredTemplateFraction:
+                doRaise = True
+            elif template_coverage < config.minTemplateFractionForExpectedSuccess:
+                doRaise = True
```

Comment:
The task only raises `NoWorkFound` if you get a RuntimeError or lsst.pex.exceptions.Exception. I didn't follow how/that you designed the input image to subtract successfully for 0.7 but not for 0.19. Maybe an inline comment to help me out. 

Also, if you designed these input images so that the 0.19 fails, then that also means that if I change the default for `minTemplateFractionForExpectedSuccess`  to 0.18, this test would fail.